internal/persistence/persistence.go

```diff
@@ -52,4 +53,10 @@ type Service interface {
 
 	// ListUsers retrieves a list of users based on the provided filter.
 	ListUsers(context.Context, *UserFilter) ([]*datatypes.User, *Page, error)
+
+	CreateBatch(context.Context, *datatypes.Batch) error
+	UpdateBatch(context.Context, uuid.UUID, BatchUpdater) (*datatypes.Batch, error)
+	DeleteBatch(context.Context, int) error
```

Comment:
I think `DeleteBatch` should use the batch UUID for selection to be consistent with the other batch methods that operate on an existing row.